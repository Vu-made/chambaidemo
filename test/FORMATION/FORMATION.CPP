#include <bits/stdc++.h>
#define N 1000001
using namespace std;
int L[N], R[N];
int n, m;
void xuly1()
{
    int i,j;
    cin >> i >> j;
    L[i] = L[j];
    R[i] = j;
    R[L[j]] = i;
    L[j] = i;
}

void xuly2()
{
    int i,j;
    cin >> i >> j;
    L[i] = j;
    R[i] = R[j];
    L[R[j]] = i;
    R[j] = i;
}

void xuly3()
{
    int i;
    cin >> i;
    R[L[i]] = R[i];
    L[R[i]] = L[i];
}

void xuly4()
{
    int i;
    cin >> i;
    cout << L[i] << ' ' << R[i] << endl;
}
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    freopen("FORMATION.INP","r",stdin);
    freopen("FORMATION.OUT","w",stdout);
    cin >> n >> m;
    for (int i=1; i<=m; i++)
    {
        int k;
        cin >> k;
        if (k == 1) xuly1();
        else if (k==2) xuly2();
        else if (k==3) xuly3();
        else xuly4();
    }
    return 0;
}
