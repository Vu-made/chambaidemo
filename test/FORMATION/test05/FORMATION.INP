63 98
1 42 1
4 1
4 1
4 1
2 52 42
1 21 42
2 15 21
1 56 52
1 25 56
1 61 52
3 56
2 32 52
1 55 15
3 21
3 25
4 15
3 55
3 52
4 15
3 32
1 5 15
4 42
2 13 42
1 22 61
3 5
2 50 15
3 13
4 15
4 61
1 12 15
3 22
3 42
2 2 61
2 11 15
3 1
2 16 61
2 4 15
1 39 15
1 48 12
1 35 4
1 18 48
4 2
4 4
3 15
1 22 39
3 22
3 16
4 2
3 11
3 39
1 38 61
3 61
3 50
1 44 18
4 35
2 15 12
1 59 48
3 44
2 1 15
1 58 2
2 31 58
4 1
3 59
2 57 18
3 1
2 51 31
3 2
3 4
4 57
1 59 38
1 23 18
1 39 38
4 48
4 51
1 4 15
4 12
4 18
2 30 57
4 18
1 43 39
1 9 59
4 43
2 16 51
3 16
2 55 51
4 15
4 4
4 18
3 59
1 44 18
4 58
4 57
4 35
1 32 57
3 18
4 32
1 54 30
3 12
