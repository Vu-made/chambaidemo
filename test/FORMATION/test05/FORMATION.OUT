42 0
42 0
42 0
55 42
0 42
15 61
0 50
22 1
16 0
35 11
61 0
12 4
15 35
18 48
57 12
31 0
48 4
23 57
23 57
59 39
4 35
12 15
23 57
38 31
18 30
15 9
44 57
